docs/gno-docs/docs/how-to-guides/write-simple-dapp.md

```diff
@@ -4,4 +4,278 @@ id: write-simple-dapp
+Our app will allow any user to create a poll, and subsequently vote 
+YAY or NAY for any poll that has not exceeded the voting deadline.
+
+## Prerequisites
+- **Text editor**
+
+## Defining dApp functionality
+
+Our dApp will consist of a Poll package, which will handle all things related to the Poll struct,
+and a Poll Factory realm, which will handle the user-facing functionality and rendering.
+
+For simplicity, we will define the functionality in plain text, and leave comments explaining the code.
+
+### Poll Package
+- Defines a `Poll` struct
+- Defines a `NewPoll` constructor
+- Defines `Poll` field getters  
+- Defines a `Vote` function
+- Defines a `HasVoted` check method
+- Defines a `VoteCount` getter method
+
+```go 
+package poll
+
+import (
+	"gno.land/p/demo/avl"
+	"std"
+)
+
+// Main struct
+type Poll struct {
+	title       string
+	description string
+	deadline    int64     // block height
+	voters      *avl.Tree // addr -> yes / no (bool)
+}
+
+// Getters
+func (p Poll) Title() string {
+	return p.title
+}
+
+func (p Poll) Description() string {
+	return p.description
+}
+
+func (p Poll) Deadline() int64 {
+	return p.deadline
+}
+
+func (p Poll) Voters() *avl.Tree {
+	return p.voters
+}
+
+// Poll instance constructor
+func NewPoll(title, description string, deadline int64) *Poll {
+	return &Poll{
+		title:       title,
+		description: description,
+		deadline:    deadline,
+		voters:      avl.NewTree(),
+	}
+}
+
+// Vote Votes for a user
+func (p *Poll) Vote(voter std.Address, vote bool) {
+	p.Voters().Set(string(voter), vote)
+}
+
+// HasVoted vote: yes - true, no - false
+func (p *Poll) HasVoted(address std.Address) (bool, bool) {
+	vote, exists := p.Voters().Get(string(address))
+	if exists {
+		return true, vote.(bool)
+	}
+	return false, false
+}
+
+// VoteCount Returns the number of yay & nay votes
+func (p Poll) VoteCount() (int, int) {
+	var yay int
+
+	p.Voters().Iterate("", "", func(key string, value interface{}) bool {
+		vote := value.(bool)
+		if vote == true {
+			yay = yay + 1
+		}
+	})
+	return yay, p.Voters().Size() - yay
+}
+```
+A few remarks:
+- We are using the `std` library for accessing blockchain-related functionality and types, such as `std.Address`.
+- Since the `map` data type is not deterministic in Go, we need to use the AVL tree structure, defined under `p/demo/avl`.
+It behaves similarly to a map; it maps a key of type `string` onto a value of any type - `interface{}`.
+- We are importing the `p/demo/avl` package directly from on-chain storage, which can be accessed through the path `gno.land/`. 
+As of October 2023, you can find already-deployed packages & libraries which provide additional Gno functionality in the [monorepo](https://github.com/gnolang/gno), under the `examples/gno.land` folder. 
+
+### Poll Factory Realm
+
+After testing & _deploying_ the Poll package, we can move on to creating the Poll Factory realm. 
```

Comment:
Please add an info section that explains you need to `deploy` the package first to import it into any Package / Realm

Reply:
Resolved in [8418fbc](https://github.com/zivkovicmilos/gno/pull/17/commits/8418fbcfdfa714e4e67c6c2d0cc996899ea267d7)